### DISCUSSION TITLE
getValues() returns undefined for one field

### DISCUSSION THREAD

**Discussion by PraveenPeri** [react-hook-form/react-hook-form#12027] [Q&A]
We have a form that is designed to use react hook form with MUI using `<Controller />`. The form renders fine and shows the initial default values correctly. If I open the form and submit it without making any changes, one field which is configured to be required is throwing the validation error though it has a value(renders as a text field with a value in it). This field is editable and when I try to retrieve the current values through `getValues()`, I see the value of this field being undefined. I am clueless on why this happening or where to start looking. Any ideas?

**Comment by tatsmaki** [comment]
It might invoke wrong onChange handler and as a result the form does not receive field updates. Please provide code-sandbox.

**Comment by PraveenPeri** [comment]
@tatsmaki Not sure I understood. The field is receiving a default value and rendering it on the screen but is throwing a required field error. There was no change made to it.  Does onChange handler still come into play? If we change the value, then it is being recognized.
I am unable to provide code-sandbox because I am not seeing this issue anywhere else. So, I am hoping if people here can just give me some pointers on where to look and debug.

**Reply by tatsmaki** [reply]
Its hard to tell something without your example. You can use `watch` to debug form state. 
```
const { watch } = useForm()

console.log(watch())

return ...
```

If it is still `undefined` then your input is not correctly connected to the form. Check your component props type definitions.
Otherwise maybe you are executing `getValues` in a wrong place.
